app/components/media/media-reactions.js

```diff
@@ -0,0 +1,54 @@
+import Component from 'ember-component';
+import get from 'ember-metal/get';
+import set from 'ember-metal/set';
+import observer from 'ember-metal/observer';
+import service from 'ember-service/inject';
+import { task } from 'ember-concurrency';
+
+export default Component.extend({
+  queryCache: service(),
+  sortOptions: [
+    '-upVotesCount',
+    '-createdAt'
+  ],
+  sort: '-upVotesCount',
+
+  refreshOnSort: observer('sort', function() {
+    get(this, 'getReactionsTask').perform();
+  }),
```

Comment:
:-1: against using observers.

If you want to re-execute this task on `sort` changing, then I recommend changing `sort` by an action and calling it there.

https://emberway.io/ember-js-functional-programming-and-the-observer-effect-48901c3b84d7